### PR TITLE
feat: resolve #169

### LINKED ISSUE #169: Implement ISR for better peformance
Target is blow pages
- nft
- order 

We can add ISR and in ISR, fetch the document data
So that we 

Prebuild is not required.

### PR DESCRIPTION
**Pull request by taijusanagi**
This is implementing ISR for better performance

### CODE REVIEW

File: packages/common/entities/nft/type.ts [hunk 1]
```diff
@@ -13,7 +14,7 @@ export interface NFTMetadata {
   animationUrl?: string;
 }
 
-export interface NFT {
+export interface NFT extends Record {
```
Comment (taijusanagi) [inline packages/common/entities/nft/type.ts:17]: Later may add updateAt comparison, so I added this.

File: packages/common/value-objects/timestamp.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export interface TimeStamp {
```
Comment (taijusanagi) [inline packages/common/value-objects/timestamp.ts:1]: This is for firebase app timestamp

File: packages/functions/src/functions/nft.ts [hunk 3]
```diff
@@ -65,21 +65,18 @@ export const sync = functions.https.onRequest(async (req, res) => {
...
-          supportsInterface,
-          holder,
-          tokenURI,
-          metadata,
-          createdAt: admin.firestore.FieldValue.serverTimestamp(),
-        },
-        { merge: true }
-      );
-    res.send({ status: true, data: "ok" });
+    const nftDoc = {
```
Comment (taijusanagi) [inline packages/functions/src/functions/nft.ts:68]: return set data too

File: packages/functions/src/functions/order.ts [hunk 4]
```diff
@@ -29,11 +29,13 @@ export const create = functions.https.onRequest(async (req, res) => {
       type,
       chainId,
       nft,
-      hash,
       isValid,
       signedOrder,
+      id: hash,
```
Comment (taijusanagi) [inline packages/functions/src/functions/order.ts:34]: standardized as hash -> id

File: packages/web/src/pages/nfts/[chainId]/[contractAddress]/[tokenId].tsx [hunk 5]
```diff
@@ -43,17 +43,28 @@ const NFTPage: NextPage<NFTPageProps> = ({ nft }) => {
 
 export default NFTPage;
 
-export const getServerSideProps: GetServerSideProps = async (context) => {
+export const getStaticProps: GetStaticProps = async (context) => {
```
Comment (taijusanagi) [inline packages/web/src/pages/nfts/[chainId]/[contractAddress]/[tokenId].tsx:46]: This is big change

This is called only
- 300 seconds passed after the last request

Comment (taijusanagi) [reply packages/web/src/pages/nfts/[chainId]/[contractAddress]/[tokenId].tsx:46]: I think I can add...

- if the record does not exists, revalidate faster

**Comment by taijusanagi** [comment]
I only add this for page which is for specific document page.

This is not applicable for the list view page.
(Because it needs to support filger)

File: packages/web/src/pages/nfts/[chainId]/[contractAddress]/[tokenId].tsx (hunk 5 above)
Comment (ksuhara) [reply packages/web/src/pages/nfts/[chainId]/[contractAddress]/[tokenId].tsx:46]: so only nft data (metadata) are synced 300 seconds each,  and order data are refreshed realtime, right?
looks good to me

**Pull request merged**